### PR TITLE
add sigmoid and gradient_descent function

### PR DESCRIPTION
**Pull request by kazourak**
add gradient descent function

### CODE REVIEW

File: srcs/logistic_regression/gradient_descent.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import pandas as pd
+import numpy as np
+from tools.sigmoid import sigmoid
+import matplotlib.pyplot as plt
+
+
+def gradient_descent(X: pd.Series, Y: pd.Series, theta: list, learning_rate: float, epoch: int) -> list:
```
Comment (farinaleo) [inline srcs/logistic_regression/gradient_descent.py]: Could be a dict

File: srcs/logistic_regression/tools/sigmoid.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import numpy as np
+import pandas as pd
+
+
+def sigmoid(z: pd.Series) -> pd.Series:
+    """
+    Compute the sigmoid function
+    :param z: float, the input
```
Comment (farinaleo) [inline srcs/logistic_regression/tools/sigmoid.py]: Doc-string deprecated, z is a pd.Series (consider as a list of elements to apply the sigmoid function)

**Review by farinaleo — CHANGES_REQUESTED** [review]
Your code is great Nolan but please change the doc-string. Thanks :)

File: srcs/logistic_regression/gradient_descent.py (hunk 1 above)
Comment (farinaleo) [reply srcs/logistic_regression/gradient_descent.py]: Thetas always start as Theta_0 = 0 and Theta_1 = 1, params are may be useless..

File: srcs/logistic_regression/gradient_descent.py [hunk 3]
```diff
@@ -4,24 +4,23 @@
...
     :param Y: pd.Series, the target
-    :param theta: list, the theta
     :param learning_rate: float, the learning rate
     :param epoch: int, the number of iterations
-    :return: list, the updated thetas
+    :return: dict, the thetas
     """
     m = len(Y)
-    thetas = theta
+    thetas = {'theta0': 0, 'theta1': 0}
```
Comment (farinaleo) [inline srcs/logistic_regression/gradient_descent.py]: theta_1 = 1

**Review by farinaleo — CHANGES_REQUESTED** [review]

**Review by farinaleo — APPROVED** [review]
You nailed it !!!!
